src/solver/simulation/sim_structure_probleme_economique.h

```diff
@@ -514,6 +545,8 @@ struct PROBLEME_HEBDO
 
     PALIERS_THERMIQUES** PaliersThermiquesDuPays = nullptr;
     ENERGIES_ET_PUISSANCES_HYDRAULIQUES** CaracteristiquesHydrauliques = nullptr;
+    std::vector<::ShortTermStorage::AREA_INPUT>* ShortTermStorage = nullptr;
```

Comment:
It turns out : no. Not here at least.

It depends on how the superseding structure is allocated. `PROBLEME_HEBDO` objects are allocated using `new`, so all all data members are properly initialized.

On the other hand, `malloc` only allocates memory, it doesn't call constructors for data members. So for example this will fail
```cpp
#include <cassert>
#include <cstdlib>

struct S
{
  int a;
  S() : a(5) {}
};

int main() {
  auto s = static_cast<S*>(malloc(sizeof(S)));
  assert(s->a == 5); // Failure
  free(s);
}
```
Objects of type `RESULTATS_HORAIRES` are allocated using `malloc`, so I got a segfault ;)